### PR TITLE
Proposal for making React >15 not complain about unknown props for elements

### PR DESCRIPTION
**Pull request by oskosk**
Updates the Card, and Popover components to now pass down extra props to HTML elements
#### Testing instructions
- Follow steps in Automattic/jetpack#4753

### CODE REVIEW

File: client/components/card/index.jsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ let Card = React.createClass( {
 	},
 
 	render: function() {
-		var { style, title, meta, icon, iconLabel, ...other } = this.props,
+		var { style, title, meta, icon, iconLabel, iconColor, compact, ...other } = this.props,
 			cardClasses = classnames( {
 				'dops-card': true,
 				'is-compact': this.props.compact
```
Comment (zinigor) [inline client/components/card/index.jsx]: Did you plan to use `compact` here instead of `this.props.compact`?

Comment (oskosk) [inline client/components/card/index.jsx]: Nope, I just wanted to take it out of what will be left in `...other`. I'll change that for some better whitelisting mechanism but I wanted to have a quick overview of what's being noisy. Looks like there were not a lot of these cases :)

File: client/components/button/index.jsx [hunk 2]
```diff
@@ -41,8 +41,10 @@ export default React.createClass( {
 			'is-scary': this.props.scary,
 			'is-borderless': this.props.borderless
 		} );
+		// Extract properties that don't apply to the HTML element
+		const { isSubmitting, isPrimary, primary, borderless, compact, ...other } = this.props;
```
Comment (psealock) [inline client/components/button/index.jsx]: use lodash `omit` here as well

**Comment by psealock** [comment]
Hi @oskosk, are you still working on this? I'd love to use some of these components and am happy to help out with the development. Cheers

**Comment by oskosk** [comment]
@psealock yes, this is still WIP but this PR was never meant to be way too comprehensive as what I did was just _fixing_ a particular case I encountered when testing React 15 in the Jetpack Admin Page (which uses dops-components).

**Pull request closed without merging**